simplic-corelib/Simplic.CoreLib/Text/StringExtraction.cs

```diff
@@ -42,5 +42,202 @@ public static class StringExtraction
+        /// </summary>
+        /// <param name="line">Line to split</param>
+        /// <param name="splitChar">Split chars</param>
+        /// <param name="minCharBeforeSplit">Minimum chars before splitting</param>
+        /// <param name="minPartLength">Minimum length</param>
+        /// <param name="cleanupChars">Chars to cleanup when length is 0</param>
+        /// <returns></returns>
+        public static IList<string> SplitLine(string line, char splitChar, int minCharBeforeSplit = 1, int minPartLength = 0, string cleanupChars = "1234567890abcdefghijklmnopqrstuvwxyz")
+        {
+            var parts = new List<string>();
+            int splitCharCount = 0;
+            var currentPart = new StringBuilder();
+
+            for (int i = 0; i < line.Length; i++)
+            {
+                var current = line[i];
+                if (splitChar == current)
+                {
+                    splitCharCount++;
+                }
+
+                if (splitCharCount >= minCharBeforeSplit)
+                {
+                    splitCharCount = 0;
+
+                    if (currentPart.Length > 0)
+                    {
+                        var part = currentPart.ToString().Trim(splitChar);
+
+                        if (!string.IsNullOrWhiteSpace(cleanupChars) && part.Length == 1)
+                        {
+                            if (!cleanupChars.Contains(part[0]))
+                                part = "";
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(part) && (minPartLength == 0 || part.Length >= minPartLength))
+                        {
+                            parts.Add(part);
+                        }
+                    }
+
+                    currentPart = new StringBuilder();
+                }
+                else
+                {
+                    if (splitChar != current)
+                        splitCharCount = 0;
+                    currentPart.Append(current);
+                }
+            }
+
+            if (currentPart.Length > 0)
+            {
+                var part = currentPart.ToString().Trim(splitChar);
+
+                if (!string.IsNullOrWhiteSpace(cleanupChars) && part.Length == 1)
+                {
+                    if (!cleanupChars.Contains(part[0]))
+                        part = "";
+                }
+
+                if (!string.IsNullOrWhiteSpace(part) && (minPartLength == 0 || part.Length >= minPartLength))
+                    parts.Add(part);
+            }
+
+            return parts;
+        }
+
+        /// <summary>
+        /// Extract a number from string and remove thousand-separator
+        /// </summary>
+        /// <param name="input">Input string</param>
+        /// <param name="separator">Separator char</param>
+        /// <returns>Extracted double or null. Throws an exception if casting failed</returns>
+        public static double CastAsNumber(string input, char separator)

```

Comment:
this whole method could be replaced with the following:

`var numbers = input.Where(Char.IsDigit)`

Reply:
As we decided, we will keep the current version.